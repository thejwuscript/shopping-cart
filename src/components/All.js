import React, { useState, useEffect } from "react";

export default function All() {
  const [games, setGames] = useState([]);

  useEffect(() => {
    async function fetchData() {
      const response = await fetch(
        "https://api.boardgameatlas.com/api/search?limit=2&client_id=R7JYD2LOOW"
      );
      const data = await response.json();
      setGames(data.games);
    }
    fetchData();
  }, []);

  return (
    <div>
      <ul>
        {games.map((game) => {
          return <li key={game.id}>{game.name}</li>;
        })}
      </ul>
    </div>
  );
}
