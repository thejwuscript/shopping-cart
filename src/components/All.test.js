import { screen, render } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { act } from "react-dom/test-utils";
import All from "./All";

const MOCK_RESPONSE = {
  games: [
    {
      id: "one",
      name: "Game One",
    },
    {
      id: "two",
      name: "Game Two",
    },
  ],
};

global.fetch = () =>
  Promise.resolve({
    json: () => Promise.resolve(MOCK_RESPONSE),
  });

describe("All", () => {
  it("renders properly", async () => {
    let component;
    await act(async () => {
      component = renderer.create(<All />);
    });
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders all games", async () => {
    render(<All />);
    expect(await screen.findByText('Game One')).toBeInTheDocument();
    expect(await screen.findByText('Game Two')).toBeInTheDocument();
  });

  // it('renders a list of games from all categories', async () => {
  //   render(<All />);
  //   expect(await screen.findByText('Game One')).toBeInTheDocument();
  //   // expect(await screen.findByText('Game Two')).toBeInTheDocument();
  // })
});
