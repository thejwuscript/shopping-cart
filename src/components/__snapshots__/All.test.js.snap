// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All renders properly 1`] = `
<div>
  <ul>
    <li>
      Game One
    </li>
    <li>
      Game Two
    </li>
  </ul>
</div>
`;
